test/CMakeLists.txt

```diff
@@ -19,25 +19,25 @@ CPMAddPackage("gh:onqtam/doctest#2.4.5")
 CPMAddPackage("gh:TheLartians/Format.cmake@1.7.0")
 
 if(TEST_INSTALLED_VERSION)
-  find_package(Greeter REQUIRED)
+  find_package(${CMAKE_PROJECT_NAME} REQUIRED)
 else()
-  CPMAddPackage(NAME Greeter SOURCE_DIR ${CMAKE_CURRENT_LIST_DIR}/..)
+  CPMAddPackage(NAME ${CMAKE_PROJECT_NAME} SOURCE_DIR ${CMAKE_CURRENT_LIST_DIR}/..)
```

Comment:
Not sure why this would work, isn't `${CMAKE_PROJECT_NAME}` the same as the project name `"GreeterTests"` in this context, as it's the top-level CMakeLists?

Reply:
IMHO you should use `${PROJECT_NAME}`

Only then it would work as **standalone** project and as **subproject** used with `CPM.cmake` in other `CMake` projects.

Reply:
@ClausKlein If test project is designed to be a standalone project then yes it may make sense to remove ${CMAKE_PROJECT_NAME}.

@TheLartians 
And regarding how it works this answer helped me to understand:
https://stackoverflow.com/questions/38938315/difference-between-cmake-project-name-and-project-name

Reply:
@DominicD yeah as I understand it, `${CMAKE_PROJECT_NAME}` should always refers to the most recent top-level project name. So it is context-dependent and we cannot know which value it holds. So if you run `cmake -Stest -Bbuild`, then `${CMAKE_PROJECT_NAME}` should be `GreeterTests`, but if you run `cmake -Sall -Bbuild`, it should be `BuildAll` etc.

I think this is also why the "Install" CI test is failing.